src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private ActionStrategy strategy;
+
+    public ShopServiceImpl(ActionStrategy strategy) {
+        this.strategy = new ActionStrategyImpl();;
+    }
```

Comment:
You pass a parameter but ignore it and create a new instance by yourself, it's not correct